internal/proto/local/plugin/host.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// Host contains all fields related to a host returned from a plugin.
+message Host {
```

Comment:
Here and for the other proto files: Why redefine `Host`, `HostSet`, and `HostCatalog` rather than use the existing protos that are already defined? That way the request object for something like `OnUpdateCatalog` above could consist of the normal HostCatalog object we already have defined (which includes attributes), along with a protobuf Struct for the secret data.

Reply:
Old commit, just resolving the conversation now.